### ISSUE TITLE
wp_nav_menu_objects expects stdClass, not an array

### ISSUE THREAD

**Issue by anugupta** [timber/timber#1865]
This recent commit - https://github.com/timber/timber/pull/1814/commits/6bc8734a24d106c48a0aa6a45ae5a3288e0b009a

uses an array for `$default_args` instead of a `stdClass` which is used in `wp-includes/nav-menu-template`

This raises a `Trying to get property  of non-object`  notice if another callback function attached to `wp_nav_menu_objects`  expects an object for `$args`

**Comment by palmiak** [comment]
@anugupta yes - you're right - thanks. We'll try to fix this.

I'm at work know but if I'm not wrong the best way it would be to do something like this:
```
// renaming $default_args to $default_args_array and than
$default_args = new stdClass();
foreach( $default_args_array as $key => $val ) {
    $default_args->$key = $val;
}
```
It looks like the best way for older PHP version. In PHP 5.4 we could do:
```
  $object = (object) [
    'propertyOne' => 'foo',
    'propertyTwo' => 42,
  ];
```

or we can just write one property after another:
```
$default_args = new stdClass();
$default_args -> menu = '';
$default_args -> container = 'div'
// etc 
```

If no one will I'll try to test it and make an PR in the evening.

**Comment by palmiak** [comment]
@gchtr  maybe you can help me a bit with this, because I see you were making this change

Now we have:
```
$menu = apply_filters( 'wp_nav_menu_objects', $menu, $default_args );
```
and $args as an array.

I have checked the `wp_nav_menu` method in WP and it looks like there:
```
$args = apply_filters( 'wp_nav_menu_args', $args );    
$args = (object) $args;
...
$sorted_menu_items = apply_filters( 'wp_nav_menu_objects', $sorted_menu_items, $args );
```

So if we would like to keep it working in Timber our code should be:
```
/**
* Default arguments from wp_nav_menu() function.
*
* @see wp_nav_menu()
*/
$default_args_array = array(
...
);

/**
* Improve compatibitility with third-party plugins.
*
* @see wp_nav_menu()
*/

$default_args = apply_filters( 'wp_nav_menu_args', $default_args_array );
$default_args = (object) $default_args;

$menu = apply_filters( 'wp_nav_menu_objects', $menu, $default_args );

$menu = self::order_children($menu);
$menu = self::strip_to_depth_limit($menu);
}
```

If I'm right I'll add a PR,

**Comment by gchtr** [comment]
@palmiak Sorry for the late answer (so busy 🤪)! I agree! I too think that we should closely follow what WordPress already does here. It would be great if you could make a pull request 👍.

**Comment by jarednova** [comment]
Closed by #1883 !
